airbyte-config/config-persistence/src/test/java/io/airbyte/config/persistence/DatabaseConfigPersistenceTest.java

```diff
@@ -82,6 +100,109 @@ void tearDown() throws Exception {
+    final StandardWorkspace workspace1 = new StandardWorkspace()
+        .withWorkspaceId(connectionId1)
+        .withCustomerId(connectionId1)
+        .withName("test-workspace")
+        .withSlug("random-string")
+        .withEmail("abc@xyz.com")
+        .withInitialSetupComplete(true)
+        .withAnonymousDataCollection(true)
+        .withNews(true)
+        .withSecurityUpdates(true)
+        .withDisplaySetupWizard(true)
+        .withTombstone(false)
+        .withNotifications(Collections.emptyList())
+        .withFirstCompletedSync(true)
+        .withFeedbackDone(true);
+    final SourceConnection sourceConnection1 = new SourceConnection()
+        .withName("source-1")
+        .withTombstone(false)
+        .withSourceDefinitionId(connectionId1)
+        .withWorkspaceId(connectionId1)
+        .withConfiguration(Jsons.emptyObject())
+        .withSourceId(connectionId1);
+
+    final DestinationConnection destinationConnection1 = new DestinationConnection()
+        .withName("destination-1")
+        .withTombstone(false)
+        .withDestinationDefinitionId(connectionId1)
+        .withWorkspaceId(connectionId1)
+        .withConfiguration(Jsons.emptyObject())
+        .withDestinationId(connectionId1);
+
+    final ResourceRequirements resourceRequirements = new ResourceRequirements()
+        .withCpuRequest("1")
+        .withCpuLimit("1")
+        .withMemoryRequest("1")
+        .withMemoryLimit("1");
+    final Schedule schedule = new Schedule().withTimeUnit(TimeUnit.DAYS).withUnits(1L);
+    final StandardSync standardSync1 = new StandardSync()
+        .withConnectionId(connectionId1)
+        .withSourceId(connectionId1)
+        .withDestinationId(connectionId1)
+        .withCatalog(new ConfiguredAirbyteCatalog())
+        .withName("standard-sync-1")
+        .withManual(true)
+        .withNamespaceDefinition(NamespaceDefinitionType.CUSTOMFORMAT)
+        .withNamespaceFormat("")
+        .withPrefix("")
+        .withResourceRequirements(resourceRequirements)
+        .withStatus(Status.ACTIVE)
+        .withSchedule(schedule);
+    final StandardSync standardSync2 = Jsons.clone(standardSync1).withConnectionId(connectionId2);
+
+    configRepository.writeStandardSourceDefinition(actorDef);
+    configRepository.writeStandardWorkspace(workspace1);
+    configRepository.writeSourceConnectionNoSecrets(sourceConnection1);
+    configRepository.writeDestinationConnectionNoSecrets(destinationConnection1);
+    // connectionId 1
+    configRepository.writeStandardSync(standardSync1);
+    configRepository.updateConnectionState(connectionId1, state);
+    final Optional<State> state1 = configRepository.getConnectionState(connectionId1);
+    final Optional<StateWrapper> state2 = statePersistence.getCurrentState(connectionId1);
+    assertEquals(state1.get().getState(), state2.get().getLegacyState().get("state"));
```

Comment:
the extra `get("state")` doesn't seem right. if this were the only issue then the `StateWrapper` could just unwrap the extra `state` field, but see the next comment...